### PR TITLE
Prefix import libraries with `lib` on mingw

### PR DESCRIPTION
**Pull request by MehdiChinoune**

### CODE REVIEW

**Comment by sdroege** [comment]
This is kind of related to https://github.com/lu-zero/cargo-c/issues/280, but only doing part of that issue. Can you update it accordingly?

CC @nirbheek

**Comment by MehdiChinoune** [comment]
> Can you update it accordingly?

update what?

File: src/build.rs [hunk 1]
```diff
@@ -232,7 +232,7 @@ fn build_implib_file(
             dlltool_command.arg("-D").arg(format!("{}.dll", name));
```
Comment (kmilos) [inline src/build.rs:232]: Prefix also the DLL to match Autotools/CMake/Meson behaviour?

File: src/build_targets.rs [hunk 2]
```diff
@@ -103,7 +103,7 @@ impl BuildTargets {
                 let impl_lib = if env == "msvc" {
                     targetdir.join(format!("{}.dll.lib", lib_name))
                 } else {
-                    targetdir.join(format!("{}.dll.a", lib_name))
+                    targetdir.join(format!("lib{}.dll.a", lib_name))
```
Comment (kmilos) [inline src/build_targets.rs:106]: Also prefix depending on msvc or not on line 102 above?

File: src/build.rs (hunk 1 above)
Comment (MehdiChinoune) [reply src/build.rs:232]: I tried to, but for some reason It failed to find the dll file when installing.

Comment (kmilos) [reply src/build.rs:232]: Perhaps because it needs to be paired w/ the change below as well?

File: src/build_targets.rs (hunk 2 above)
Comment (MehdiChinoune) [reply src/build_targets.rs:106]: What do you?
If env==msvc import library has an output name `${libname}.dll.lib`, else (==gnu) it has `lib${libname}.dll.a`

File: src/build.rs (hunk 1 above)
Comment (MehdiChinoune) [reply src/build.rs:232]: I tried that too, You could try it yourself. Patch cargo-c As I did and see if it works for you.

File: src/build_targets.rs (hunk 2 above)
Comment (kmilos) [reply src/build_targets.rs:106]: Not only `imp_lib` here, but also `shared_lib` on line 102 above

**Comment by kmilos** [comment]
@MehdiChinoune Indeed, `{}.dll` is produced no matter what, as if cargo itself needs to be patched...

**Comment by kmilos** [comment]
Ah, perhaps also adding to https://github.com/lu-zero/cargo-c/blob/7760ca3d4cb31a7d8a3cbfe003576e2f47acd7f7/src/target.rs#L107-L110

**Pull request closed without merging**